src/main/java/hudson/plugins/testlink/TestLinkBuilder.java

```diff
@@ -159,120 +159,131 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
 
 		TestLinkHelper.setTestLinkJavaAPIProperties(installation.getTestLinkJavaAPIProperties(), listener);
 
-		final TestLinkSite testLinkSite;
-		final TestCaseWrapper[] automatedTestCases;
+		TestLinkSite testLinkSite;
+		TestCaseWrapper[] automatedTestCases;
 		final String testLinkUrl = installation.getUrl();
 		final String testLinkDevKey = installation.getDevKey();
 		listener.getLogger().println(Messages.TestLinkBuilder_UsedTLURL(testLinkUrl));
 
-		try {
 			final String testProjectName = expandVariable(build.getBuildVariableResolver(),
 					build.getEnvironment(listener), getTestProjectName());
 			final String testPlanName = expandVariable(build.getBuildVariableResolver(),
 					build.getEnvironment(listener), getTestPlanName());
-			final String platformName = expandVariable(build.getBuildVariableResolver(),
+			String platformName;
+			final String readPlatforms = expandVariable(build.getBuildVariableResolver(),
 					build.getEnvironment(listener), getPlatformName());
 			final String buildName = expandVariable(build.getBuildVariableResolver(),
 					build.getEnvironment(listener), getBuildName());
 			final String buildNotes = Messages.TestLinkBuilder_Build_Notes();
-			if(LOGGER.isLoggable(Level.FINE)) {
-				LOGGER.log(Level.FINE, "TestLink project name: ["+testProjectName+"]");
-				LOGGER.log(Level.FINE, "TestLink plan name: ["+testPlanName+"]");
-				LOGGER.log(Level.FINE, "TestLink platform name: ["+platformName+"]");
-				LOGGER.log(Level.FINE, "TestLink build name: ["+buildName+"]");
-				LOGGER.log(Level.FINE, "TestLink build notes: ["+buildNotes+"]");
+
+			int iterateCount=1; // Multiplatform support logic
+			String [] platformsAll = null;
+			if (readPlatforms != null){

```

Comment:
One thing that I want to test is that it still works when you don't have any platform. Have you tested it @yasassri ?